pkg/chunk/cache/redis_client.go

```diff
@@ -0,0 +1,304 @@
+	PoolSize    int            `yaml:"pool_size"`
+	Password    flagext.Secret `yaml:"password"`
+	EnableTLS   bool           `yaml:"enable_tls"`
+	IdleTimeout time.Duration  `yaml:"idle_timeout"`
+	MaxConnAge  time.Duration  `yaml:"max_connection_age"`
+
+	DeprecatedMaxIdleConns         int           `yaml:"max_idle_conns"`
+	DeprecatedMaxActiveConns       int           `yaml:"max_active_conns"`
+	DeprecatedMaxConnLifetime      time.Duration `yaml:"max_conn_lifetime"`
+	DeprecatedWaitOnPoolExhaustion bool          `yaml:"wait_on_pool_exhaustion"`
+}
+
+// RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet
+func (cfg *RedisConfig) RegisterFlagsWithPrefix(prefix, description string, f *flag.FlagSet) {
+	f.StringVar(&cfg.Topology, prefix+"redis.topology", redisTopologyServer, description+"Redis topology. Supported: "+redisTopologyServer+", "+redisTopologyCluster+", "+redisTopologySentinel+".")
+	f.StringVar(&cfg.Endpoint, prefix+"redis.endpoint", "", description+"Redis service endpoint to use when caching chunks. If empty, no redis will be used.")
+	f.DurationVar(&cfg.Timeout, prefix+"redis.timeout", 100*time.Millisecond, description+"Maximum time to wait before giving up on redis requests.")
+	f.DurationVar(&cfg.Expiration, prefix+"redis.expiration", 0, description+"How long keys stay in the redis.")
+	f.IntVar(&cfg.PoolSize, prefix+"redis.pool-size", 0, description+"Maximum number of connections in the pool.")
+	f.Var(&cfg.Password, prefix+"redis.password", description+"Password to use when connecting to redis.")
+	f.BoolVar(&cfg.EnableTLS, prefix+"redis.enable-tls", false, description+"Enables connecting to redis with TLS.")
+	f.DurationVar(&cfg.IdleTimeout, prefix+"redis.idle-timeout", 0, description+"Close connections after remaining idle for this duration. If the value is zero, then idle connections are not closed.")
+	f.DurationVar(&cfg.MaxConnAge, prefix+"redis.max-connection-age", 0, description+"Close connections older than this duration. If the value is zero, then the pool does not close connections based on age.")
+
+	f.IntVar(&cfg.DeprecatedMaxIdleConns, prefix+"redis.max-idle-conns", 0, "Deprecated: "+description+"Maximum number of idle connections in pool.")
+	f.IntVar(&cfg.DeprecatedMaxActiveConns, prefix+"redis.max-active-conns", 0, "Deprecated (use pool-size instead): "+description+"Maximum number of active connections in pool.")
+	f.DurationVar(&cfg.DeprecatedMaxConnLifetime, prefix+"redis.max-conn-lifetime", 0, "Deprecated (use max-connection-age instead): "+description+"Close connections older than this duration.")
+	f.BoolVar(&cfg.DeprecatedWaitOnPoolExhaustion, prefix+"redis.wait-on-pool-exhaustion", false, "Deprecated: "+description+"Enables waiting if there are no idle connections. If the value is false and the pool is at the max-active-conns limit, the pool will return a connection with ErrPoolExhausted error and not wait for idle connections.")
```

Comment:
Given Redis support is still marked as experimental, I would suggest to remove all such deprecated config options (mention it in the CHANGELOG, please). We can introduce config breaking changes for any experimental feature.